group-coordinator/src/main/java/org/apache/kafka/coordinator/group/modern/SubscribedTopicDescriberImpl.java

```diff
@@ -74,6 +83,30 @@ public Set<String> racksForPartition(Uuid topicId, int partition) {
         return Set.of();
     }
 
+    /**
+     * Returns a set of assignable partitions from the topic metadata.
+     * If the allowed partition map is null, all the partitions in the corresponding
+     * topic metadata are returned for the argument topic id. If allowed map is empty,
+     * empty set is returned.
+     *
+     * @param topicId The uuid of the topic
+     * @return Set of integers if assignable partitions available, empty otherwise.
+     * @throws UnknownTopicIdException if the topicId is not found in the metadata.
```

Comment:
I do not think you should throw `UnknownTopicIdException` here. The partition assignor will only throw `PartitionAssignorException` so this is likely a bit inconvenient. I would return `null` for a missing topic ID, an empty set if the topic ID is known but there are no assignable partitions, and the set of partition indices for the assignable partitions.